src/supertux/menu/sorted_contrib_menu.cpp

```diff
@@ -38,10 +43,83 @@ SortedContribMenu::SortedContribMenu(std::vector<std::unique_ptr<World>>& worlds
     {
       m_world_folders.push_back(worlds[i]->get_basedir());
       std::string title_str;
+      const auto savegame = Savegame::from_file(worlds[i]->get_savegame_filename());
+
       if (worlds[i]->is_levelset())
-        title_str = "[" + worlds[i]->get_title() + "]";
+      {
+        uint32_t level_count = 0, solved_count = 0;
+        const auto& state = savegame->get_levelset_state(worlds[i]->get_basedir());
+
+        for (const auto& level_state : state.level_states)
+        {
+          if (level_state.filename.empty() || level_state.filename.back() == '~') continue;
+          if (level_state.solved) ++solved_count;
+          ++level_count;
+        }
+
+        if (!level_count)
+        {
+          title_str = str(boost::format(_("[%s] *NEW*")) % worlds[i]->get_title());
+        }
+        else
+        {
+          title_str = str(boost::format(_("[%s] (%u/%u; %u%%)")) % worlds[i]->get_title() %
+                          solved_count % level_count % (100 * solved_count / level_count));
+        }
+      }
       else
-        title_str = worlds[i]->get_title();
+      {
+        uint32_t island_level_count = 0, island_solved_count = 0,
+                 world_level_count = 0,  world_solved_count = 0;
+        std::string wm_filename = savegame->get_player_status().last_worldmap;
+
+        if (wm_filename.empty())
+          wm_filename = worlds[i]->get_worldmap_filename();
+
+        wm_filename = "/" + wm_filename;
+
+        const auto& state = savegame->get_worldmap_state(wm_filename);
+        for (const auto& level_state : state.level_states)
+        {
+          if (level_state.filename.empty()) continue;
+          if (level_state.solved) ++island_solved_count;
+          ++island_level_count;
+        }
+
+        const auto levelset = std::unique_ptr<Levelset>(new Levelset(worlds[i]->get_basedir(), true));
+        world_level_count = levelset->get_num_levels();
+        const auto world_list = savegame->get_worldmaps();
+
+        for (const auto& world : world_list)
+        {
+          const auto& world_state = savegame->get_worldmap_state(world);
+          for (const auto& level_state : world_state.level_states)
+          {
+            if (level_state.filename.empty()) continue;
+            if (level_state.solved) ++world_solved_count;
+          }
+        }
```

Comment:
Same as before, but for `WorldmapState` as well. The code is being repeated at this point.